src/main/java/com/twilio/jwt/validation/SignatureCreator.java

```diff
@@ -0,0 +1,199 @@
+package com.twilio.jwt.validation;
+
+import com.twilio.exception.InvalidRequestException;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.hash.HashFunction;
+
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Creates a canonical signature out of HTTP request components.
+ * <p>
+ * The process of generating the signature is described <a href="https://www.twilio
+ * .com/docs/api/credentials/public-key-client-validation-getting-started#3-create-hash-of-the-canonical-request"
+ * >here</a>.
+ */
+class SignatureCreator {
```

Comment:
I agree that this might not be the best name, should we call this CanonicalRequestCreator?

@jingming thoughts on this? or keep it as it is?